test/integration/shell_int_test.go

```diff
@@ -276,6 +280,64 @@ func (suite *ShellIntegrationTestSuite) SetupRCFile(ts *e2e.Session) {
 	suite.Require().NoError(err)
 }
 
+func (suite *ShellIntegrationTestSuite) TestNestedShellNotification() {
+	if runtime.GOOS == "windows" {
+		return // cmd.exe does not have an RC file to check for nested shells in
+	}
+	suite.OnlyRunForTags(tagsuite.Shell)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	cfg, err := config.New()
+	suite.Require().NoError(err)
+
+	// Setup subshell. Since our CI tests are running in bash, override with zsh on macOS.
+	var ss subshell.SubShell
+	env := []string{"ACTIVESTATE_CLI_DISABLE_RUNTIME=false"}
+	if runtime.GOOS != "darwin" || !condition.OnCI() {
+		ss = subshell.New(cfg)
+	} else {
+		cfg.Set(subshell.ConfigKeyShell, "zsh")
```

Comment:
Cannot set SHELL env here via `os.Setenv()` because that might interfere with parallel tests.